camera_manager.py

```diff
@@ -0,0 +1,154 @@
+        cameras = []
+        for camera_name, camera_path in camera_dict.items():
+            camera = Camera(camera_name, camera_path)
+            cameras.append(camera)
+        return cameras
+    
+    def __get_camera(self, camera_name: str) -> Camera:
+        """
+        Get the Camera object associated with a given camera name
+        """
+        # TODO: Raise error if camera not found
+        for camera in self.cameras:
+            if camera.name == camera_name:
+                return camera
+    
+    def get_camrea_fps(self, camera_name: str)-> int:
+        """
+        Return the fps for a camera, given a camera name
+        """
+        # TODO: Raise error if camera not found
+        camera = self.__get_camera(camera_name)
+        return camera.fps
+    
+    def set_camera_fps(self, camera_name: str, fps: int):
+        """
+        Change the fps for a camera, given a camera name and fps
+        """
+        # TODO: Raise error if camera not found
+        camera = self.__get_camera(camera_name)
+        camera.fps = fps
+
+    def get_camera_encoding_params(self, camera_name: str):
+        """
+        Return the encoding parameters given a camera name
+        """
+        # TODO: Raise error if camera not found
+        camera = self.__get_camera(camera_name)
+        return camera.encoding_params
+        
+    def set_camera_encoding_params(self, camera_name: str, encoding_quality: int):
+        """
+        Set the camera encoding parameters given a camera name
+        """
+        # TODO: Raise error if camera not found
+        camera = self.__get_camera(camera_name)
+        camera.encoding_params[1] = encoding_quality
+
+    def camera_is_running(self, camera_name: str) -> bool:
+        """
+        Return True if camera stream is being asked for and False if it has been ended
+        """
+        # TODO: Raise error if camera not found
+        camera = self.__get_camera(camera_name)
+        return camera.is_running
+
+    
+    def start_video_capture(self, camera_name: str) -> cv2.VideoCapture:
+        """
+        Given a camera name, return an openCV video capture object (to read frames)
+        """
+        # TODO: Raise error if camera not found
+        camera = self.__get_camera(camera_name)
+        # TODO: Check if camera is already running
+        cap = cv2.VideoCapture(camera.path)
+        camera.is_running = True
+        return cap
+    
+    def end_video_capture(self, camera_name: str):
+        """
+        Set a camera to no longer run
+        """
+        # TODO: Raise error if camera not found
+        camera = self.__get_camera(camera_name)
+        camera.is_running = False
+
+    def get_available_cameras(self) -> list[str]:
+        """
+        Return a list of cameras which are not currently being used
+        """
+        available_cameras = []
+        for camera in self.cameras:
+            if camera.is_running == False:
```

Comment:
ruff caught a pep8 violation: https://docs.astral.sh/ruff/rules/true-false-comparison/

```suggestion
            if camera.is_running is False:
```

Reply:
Should be fixed

Reply:
thanks! note that my suggestion was wrong. `is` compares the singular boolean objects, but idiomatic Python is expected to use `if my_bool:` or `if not my_bool`. 

i like your solution here! 😄